### DISCUSSION TITLE
Semantic-release sync bump version in two branches

### DISCUSSION THREAD

**Discussion by Camisado** [semantic-release/semantic-release#3519] [Q&A]
Hi, semantic-release community!

Let me describe the thing.
I have a library repo with two main branches: `development` and `main`.
`development` branch is for daily dev work, and it collects all the changes from feature branches.
`main` branch is used for from-time-to-time releases: updates are cutt-off from `development` branch into `release/***` branch, and  later get merged into `main`.

I have CI/CD pipeline for main branch to publish package in registry by `semantic-release` with this config:

```
{
  "branches": ["main"],
  "plugins": [
    "@semantic-release/commit-analyzer",
    "@semantic-release/release-notes-generator",
    "@semantic-release/gitlab",
    [
      "@semantic-release/npm",
      {
        "assets": [ "dist/**/*", "package.json" ]
      }
    ],
    [
      "@semantic-release/git",
      {
        "assets": ["package.json"],
        "message": "chore(release): ${nextRelease.version} [skip ci]\n\n${nextRelease.notes}"
      }
    ]
  ]
}
```

Basically it works like a charm, and as expected get version bump commit in `main` branch.
But in the `development` branch I still have the previous version in package.json, and when I'll do next publish from `development` branch into `main` through `release/***` I'll get incorrect new version. It sounds like a trivial case, but didn't find trivial solutions according to `semantic-release` functionality.

So, my main question what is a legit way to sync version bump between `main` and `development` branches?

**Comment by travi** [comment]
semantic-release is designed for a continuous-delivery style workflow and does not officially support gitflow or similar variations. You are welcome to search the history of support issues for how others have accomplished partial super, but what you describe is not a workflow that we intend to add official support for or provide help to accomplish without official support.
